libflux/src/core/parser/mod.rs

```diff
@@ -532,6 +532,88 @@ impl Parser {
             id,
         }))
     }
+    #[cfg(test)]
+    fn parse_type_expression(&mut self) -> TypeExpression {
+        let monotype = self.parse_monotype(); // monotype
+
+        // let t = self.peek();
+        // if t.tok == TOK_IDENT {
+        //     t = self.expect(TOK_INDENT); // where
+        //     if t.lit != "where" {
+        //         // handle error if where was not found
+        //         self.errs.push(format!(
+        //             "expected \"where\" but found : {}",
+        //             t.lit
+        //         ));
+        //     }
+        //
+        //     let constraints = self.parse_constraints();
+        //     // return the typeexpression
+        //     Expression::TypeExpression {
+        //         Monotype: monotype,
+        //         Constraints: constraints
+        //     }
+        // }
+        // return the typeexpression
+        TypeExpression{
+            monotype: monotype.clone(),
+            // constraints: None
+            base: base_from_monotype(&monotype) ,
+        }
+    }
+
+    #[cfg(test)]
+    fn parse_monotype(&mut self) -> MonoType {
+        // Tvar | Basic | Array | Record | Function
+
+        // worry about Basic only rn.
+        let t = self.peek();
+        match t.tok {
+            TOK_IDENT => if t.lit.len() == 1 { self.parse_tvar() }
+            else { self.parse_basic() }
```

Comment:
I don't think this is formatted correctly. You can run `rustfmt` to format your rust files. Optionally you can also run `make fmt` which includes `rustfmt`.

Reply:
✅